catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/cache/solr/impl/CachingFederationStrategyTest.java

```diff
@@ -169,29 +169,90 @@ public void setup() throws Exception {
     strategy.setCacheCommitPhaser(cacheCommitPhaser);
     strategy.setCacheBulkProcessor(cacheBulkProcessor);
 
-    mockQuery = mock(Query.class);
     when(mockQuery.getTimeoutMillis()).thenReturn(LONG_TIMEOUT);
     when(mockQuery.getPageSize()).thenReturn(-1);
 
-    mockMetacard = new MetacardImpl();
-    mockMetacard.setId("mock metacard");
+    metacard = new MetacardImpl();
+    metacard.setId("mock metacard");
 
-    mockResult = new ResultImpl(mockMetacard);
+    Result mockResult = new ResultImpl(metacard);
 
-    mockResponse = mock(SourceResponse.class);
-    results = Arrays.asList(mockResult);
+    List<Result> results = Arrays.asList(mockResult);
     when(mockResponse.getResults()).thenReturn(results);
 
     // Set general properties
     properties = new HashMap<>();
-    properties.put(QUERY_MODE, NATIVE_QUERY_MODE);
   }
 
   @After
   public void tearDownClass() throws Exception {
     strategy.shutdown();
   }
 
+  @Test
+  public void testFederateGetEmptyResults() throws Exception {
+    List<String> sourceIds = ImmutableList.of("1");
+
+    QueryRequest fedQueryRequest = new QueryRequestImpl(mockQuery, false, sourceIds, properties);
+
+    List<Source> sourceList = ImmutableList.of();
+
+    when(mockResponse.getHits()).thenReturn((long) sourceList.size());
+
+    QueryResponse federateResponse = federateStrategy.federate(sourceList, fedQueryRequest);
+
+    assertThat(federateResponse.getResults().size(), is(0));
+  }
+
+  @Test
+  public void testFederateGetResults() throws Exception {
```

Comment:
❓ This test doesn't make sense to me. It's asserting that `getResults().size() == sourceList.size()`. I understand why both equal 3 because of how `getMockSource()` works, but it's still conflating several different numbers.